### DISCUSSION TITLE
SEO Title/Description null for collection in storefront API

### DISCUSSION THREAD

**Discussion by justinhenricks** [Shopify/storefront-api-feedback#181] [Help] [answered]
### Steps to Reproduce

1. Create a new collection in shopify admin with a Title and Description
2. Query graphql storefront api 'collection' with seo:
```
query Collection($handle:String!){
  collection(handle:$handle){
    id
    title
    seo {
      title
      description
    }
  }
}
```

seo title and description are null. 

If you go back to the collection in Shopify admin and explicitly open up the SEO settings and change the title/description fields and save and try again, then the fields come down correctly.

### Expected Behavior

seo title and description are populated correctly upon creating a new collection without having to open up the SEO section on the collection edit page and manually change/save those values.

### Actual Behavior

SEO title and description are null until you go and change the values explicitly in the shopify admin edit collection

**Comment by johncraigcole** [comment, marked as answer]
Hey @justinhenricks sorry for the slow reply here. This appears to be the behavior the Admin has had for some time now. I'd have to dig into the why, perhaps this around not duplicating data as we won't store an SEO title or description that match the resources title or description. 

On online store (liquid), or in Hydrogen we fallback to the resource title or description if `seo.title` or `seo.description` are null. So for now, if you're building a custom/headless storefront with the Storefront API I would follow the same pattern.

**Reply by justinhenricks** [reply]
Hey, @johncraigcole thanks for the reply - that's reasonable. I believe the current storefront API schema says seo title and description are non-null though, perhaps that should be updated.

**Reply by johncraigcole** [reply]
The `seo` field itself on say a product, yes is [non null](https://shopify.dev/docs/api/storefront/2023-04/objects/Product#field-product-seo), but the [fields of type SEO (title, and description) are nullable](https://shopify.dev/docs/api/storefront/2023-04/objects/SEO)

**Reply by justinhenricks** [reply]
Got it - I thought a String without ? or something was nullable but I do see the non-null descriptor now. Thanks!

**Comment by dev-erich** [comment]
Has this been resolved yet? It appears to be the same issue for Products as well, really anything using these SEO fields... Nearly 2 years now and still no update

**Comment by jasonloeve** [comment]
Been running into this issue too, how is this not resolved yet ?

Just switched Hydrogen storefronts, query collections

`query collection(
  $handle: String!
  $country: CountryCode
  $language: LanguageCode
  $first: Int
  $last: Int
  $startCursor: String
  $endCursor: String
) @inContext(country: $country, language: $language) {
  collection(handle: $handle) {
    id
    handle
    title
    description
    seo {
      description
      title
    }
    products(
      first: $first,
      last: $last,
      before: $startCursor,
      after: $endCursor
    ) {
      nodes {
        ...BasicProduct
      }
      pageInfo {
        hasPreviousPage
        hasNextPage
        endCursor
        startCursor
      }
    }
  }
}`

seo.description and  seo.description return null even though there is a seo description and title saved, edit description and retest, response now returns the description but title is null, edit title by adding just a space and save seo title is now returned.

I wouldnt care manually updating collections if i had a small handful but i have a few hundred collections that already have this data.
